File: library/src/main/java/com/phantomvk/identifier/interfaces/Disposable.java
package com.phantomvk.identifier.interfaces;

/**
 * Represents a disposable resource.
 */
public interface Disposable {
  /**
   * Dispose the resource, the operation should be idempotent.
   */
  void dispose();
  
  /**
   * Returns true if this resource has been disposed.
   *
   * @return true if this resource has been disposed
   */
  boolean isDisposed();
}

File: library/src/main/java/com/phantomvk/identifier/interfaces/Disposable.kt
package com.phantomvk.identifier.interfaces

/**
 * Represents a disposable resource.
 */
interface Disposable {
  /**
   * Dispose the resource, the operation should be idempotent.
   */
  fun dispose()

  /**
   * Returns true if this resource has been disposed.
   * @return true if this resource has been disposed
   */
  fun isDisposed(): Boolean
}
